### PR TITLE
Scottx611x/galaxy cleanup on delete or cancel

### LINKED ISSUE #2375: Galaxy objs. persist when deleting/cancelling running Analyses
* Specific code commit: 7e8847fb36e4a64352247def0ff60fad095abcef
* Version of the web browser and OS: `Chrome Ubuntu 16.04`
* Environment where the error occurred (Vagrant VM and site conf mode or AWS instance): 
`test.stemcellcommons.org`

### Steps to reproduce
1. Run an Analysis
2. Cancel or Delete said _**running**_ Analysis

### Observed behavior
* If the`REFINERY_GALAXY_ANALYSIS_CLEANUP` setting is set to `on_success` then these Galaxy Artifacts are only cleaned up when an Analysis is successful. Analyses are not deemed to be successful if we cancel or delete them.

### Expected behavior
* Even if the the `REFINERY_GALAXY_ANALYSIS_CLEANUP` setting is set to: `on_success`, deleting or cancelling an Analysis should remove it's related Galaxy artifacts.

### General Notes:
* This is not an issue if: `REFINERY_GALAXY_ANALYSIS_CLEANUP` is set to `always`

### PR DESCRIPTION
**Pull request by scottx611x**
Fixes #2375

### CODE REVIEW

**Comment by scottx611x** [comment]
Build failure is known. See : #2348

File: refinery/core/models.py [hunk 1]
```diff
@@ -1309,17 +1309,22 @@ def galaxy_progress(self):
...
         return history['percent_complete']
 
-    def galaxy_cleanup(self):
-        """Determine when/if Galaxy libraries, workflows and histories are
-        to be deleted based on the value of
-        global setting: REFINERY_GALAXY_ANALYSIS_CLEANUP"""
-
-        cleanup = settings.REFINERY_GALAXY_ANALYSIS_CLEANUP
-
-        if (cleanup == 'always' or
```
Comment (hackdna) [inline refinery/core/models.py]: Why not simply add a check for `self.canceled` here?

File: refinery/galaxy_connector/models.py [hunk 2]
```diff
@@ -53,3 +59,16 @@ def get_history_file_list(self, history_id):
 
             files.append(file_info)
         return files
+
+    def delete_history(self, history_id):
+        try:
+            self.galaxy_connection().histories.delete_history(history_id,
+                                                              purge=True)
+        except galaxy.client.ConnectionError as e:
+            logger.error(error_msg, 'history', self.name, e.message)
```
Comment (scottx611x) [inline refinery/galaxy_connector/models.py]: Fix this

**Comment by scottx611x** [comment]
ping @hackdna

**Pull request merged**